### ISSUE TITLE
Forced to use async filter functions (regression/breaking change)

### ISSUE THREAD

**Issue by Arthy000** [DevExpress/testcafe#6705]
### What is your Scenario?

Be able (again) to use filter functions that return a boolean.

### What is the Current behavior?

When working with TypeScript, filter functions HAVE to return a Promise<boolean>
Before #6370 #6371 it had to return a boolean.
(issues occurred when I upgraded from 1.15 to 1.17)
This prevents TypeScript compilation.

### What is the Expected behavior?

As stated in the description of #6371 I should have a choice between sync (boolean) and async (Promise<boolean>)
Otherwise the change breaks filter functions that used to work.

### What is your public website URL? (or attach your complete example)

N/A

### What is your TestCafe test code?

```ts
const filterTests = (scenario?: string) => (scenarioName: string) =>
  !scenario || !!scenarioName.match(new RegExp(scenario, 'i'));

return runner.filter(filterTests(testName));
```
Had to be changed to
```ts
const filterTests = (scenario?: string) => async (scenarioName: string) =>
  !scenario || !!scenarioName.match(new RegExp(scenario, 'i'));

return runner.filter(filterTests(testName));
```

That makes little sense since nothing is async here

### Screenshots

![image](https://user-images.githubusercontent.com/1800622/142411359-cae97890-033a-4170-9fe1-48108883e807.png)

### Steps to Reproduce

N/A

### TestCafe version

1.17.1

### Command-line arguments

N/A

### Other

Fixing it should be easy
```diff
type FilterFunction = (
    testName: string,
    fixtureName: string,
    fixturePath: string,
    testMeta: Metadata,
    fixtureMeta: Metadata
- ) => Promise<boolean>
+ ) => Promise<boolean> | boolean;
```

But at the moment I can't test this to make sure I'm not breaking anything, and I'd rather not submit untested code

**Comment by AlexKamaev** [comment]
I was able to reproduce the issue. Thank you for reporting this.

**Comment by miherlosev** [comment]
Fixed in the `testcafe@1.17.2-rc.2`.
